models/org_team.go

```diff
@@ -264,6 +279,69 @@ func (t *Team) AddRepository(repo *Repository) (err error) {
 	return sess.Commit()
 }
 
+// RemoveAllRepositories removes all repositories from team and recalculates access
+func (t *Team) RemoveAllRepositories() (err error) {
+	if t.IncludesAllRepositories {
+		return nil
+	}
+
+	sess := x.NewSession()
+	defer sess.Close()
+	if err = sess.Begin(); err != nil {
+		return err
+	}
+
+	if err = t.removeAllRepositories(sess); err != nil {
+		return err
+	}
+
+	t.NumRepos = 0
+	if _, err = sess.ID(t.ID).Cols("num_repos").Update(t); err != nil {
+		return err
+	}
+
+	return sess.Commit()
+}
+
+// removeAllRepositories removes all repositories from team and recalculates access
+// Note: Shall not be called if team includes all repositories
+func (t *Team) removeAllRepositories(e Engine) (err error) {
+	// Delete all accesses.
+	for _, repo := range t.Repos {
+		if err := repo.recalculateTeamAccesses(e, t.ID); err != nil {
+			return err
+		}
+
+		// Remove watches from all users and now unaccessible repos
+		for _, user := range t.Members {
+			has, err := hasAccess(e, user.ID, repo)
+			if err != nil {
+				return err
+			} else if has {
+				continue
+			}
+
+			if err = watchRepo(e, user.ID, repo.ID, false); err != nil {
+				return err
+			}
+
+			// Remove all IssueWatches a user has subscribed to in the repositories
+			if err = removeIssueWatchersByRepoID(e, user.ID, repo.ID); err != nil {
```

Comment:
Why remove user from issues they explicitly watch?

Reply:
It is only removed if the user doesn't have any access left to the repo. It is the same as when a single repo is removed.